private/base.rkt

```diff
@@ -0,0 +1,223 @@
+#lang at-exp racket/base
+(require racket/contract
+         racket/dict
+         racket/format
+         racket/file
+         racket/path
+         compiler/compilation-path
+         compiler/cm
+         "exn-gobbler.rkt")
+
+(provide log-quickscript-fatal
+         log-quickscript-error
+         log-quickscript-warning
+         log-quickscript-info
+         log-quickscript-debug
+         quickscript-logger
+         get-script-help-string
+         make-simple-script-string
+         prop-dict-ref
+         compile-user-scripts
+         compile-user-script
+         this-os-type
+         time-info
+         path-free?
+         path-string=?
+         script-file?
+         user-script-dir
+         quickscript-dir
+         get-property-dicts
+         path-string->string)
+
+(module+ test
+  (require rackunit))
+
+(define version-bytes (string->bytes/utf-8 (version)))
+(define vm-bytes (string->bytes/utf-8 (symbol->string (system-type 'vm))))
+
+(define-logger quickscript)
+
+;; TODO: What if (find-system-path 'pref-dir) does not exist?
+(define quickscript-dir
+  (or (getenv "PLTQUICKSCRIPTDIR")
+      (build-path (find-system-path 'pref-dir) "quickscript")))
+
+(define user-script-dir
+  (build-path quickscript-dir "user-scripts"))
+
+(define (path-free? p-str)
+  (not (path-only p-str)))
+
+(define (path-string->string p-str)
+  (if (string? p-str)
+      p-str
+      (path->string p-str)))
+
+(define (script-file? f)
+  (and (equal? (path-get-extension f) #".rkt")
+       (not (equal? f (if (string? f)
+                          "info.rkt"
+                          (string->path-element "info.rkt"))))))
```

Comment:
This special-case for `info.rkt` feels a bit unsatisfying, but it's definitely the easiest approach.

Reply:
Then we should also prevent the user from creating a script named `info` on the gui:
https://github.com/Metaxal/quickscript/blob/master/tool.rkt#L127

Reply:
Good catch!

Reply:
I've updated `(new-script)`, though I'm a bit dubious of its overall approach.